### PR TITLE
Infra (CI/CD): Do dry run (fake) deployment to testing subdirectory in formidable.com bucket.

### PR DESCRIPTION
**Pull request by ryan-roemer**
## Work

- Fix the routes build logic to properly produce `docs` sidebar.
- Add docs checks + build to CI.
- Does a `--dryrun` simulated run on merging to a temporary `open-source/spectacle-TODO-TESTING` subdirectory in real production s3 bucket.

## Tasks to complete

- [x] Ticket or implement a staging per-PR deploy step. Could be netlify or surge.
- [x] Add a `deploy` step that at least echoes something or maybe does an AWS `--dryrun` to the formidable s3 bucket.
- [x] Deploy to something like `https://formidable.com/open-source/spectacle-TEMP/` to check if it comes up (it will be somewhat broken because of routes but we'd just check that files got there, not that they'd work).
- [x] Ticket out remaining work needed (in `formidable.com` repo and in this one).

... rest of tickets captured to https://github.com/FormidableLabs/spectacle/issues/793

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -48,6 +48,35 @@ Then visit: http://localhost:3000/open-source/spectacle/
...
+First, install the AWS CLI:
+
+```sh
+$ brew install awscli
+```
+
+Then, set up `aws-vault` with the AWS access and secret keys for `spectacle` CI in the `AWS IAM (formidable-com)` entry in the IC vault:
+
+```sh
+$ brew install aws-vault
```
Comment (boygirl) [inline docs/README.md]: When going through these steps I was prompted to `brew cask install aws-vault`. Should this line be changed?

Comment (ryan-roemer) [reply docs/README.md]: Yep! Good catch!

**Comment by boygirl** [comment]
@ryan-roemer I went through the dryrun described in the docs readme, and everything works well for me.

**Comment by kale-stew** [comment]
I was prompted to set a new password for the keychain after entering the two keys. 🤔 Is this expected behavior?

Screen Shot 2020-01-09 at 4 00 31 PM

**Comment by ryan-roemer** [comment]
@boygirl I went ahead and did the `surge` deployment addition and it looks good. Current staging site is available at:

https://formidable-com-spectacle-staging-791.surge.sh/open-source/spectacle/

**Pull request merged**